Tweetinvi.Controllers/Messages/MessageController.cs

```diff
@@ -12,96 +14,120 @@ public class MessageController : IMessageController
-            return _messageFactory.GenerateMessagesFromMessagesDTO(messagesDTO);
+            return GetLatestMessages(count, out _);
         }
 
-        public IEnumerable<IMessage> GetLatestMessagesSent(int maximumMessages = TweetinviConsts.MESSAGE_GET_COUNT)
+        public IEnumerable<IMessage> GetLatestMessages(int count, out string cursor)
         {
-            var parameter = new MessagesSentParameters
+            var parameters = new GetMessagesParameters()
             {
-                MaximumNumberOfMessagesToRetrieve = maximumMessages
+                Count = count
             };
 
-            var messagesDTO = _messageQueryExecutor.GetLatestMessagesSent(parameter);
-            return _messageFactory.GenerateMessagesFromMessagesDTO(messagesDTO);
+            return GetLatestMessages(parameters, out cursor);
         }
 
-        public IEnumerable<IMessage> GetLatestMessagesSent(IMessagesSentParameters messagesSentParameters)
+        public IEnumerable<IMessage> GetLatestMessages(IGetMessagesParameters parameters)
         {
-            var messagesDTO = _messageQueryExecutor.GetLatestMessagesSent(messagesSentParameters);
-            return _messageFactory.GenerateMessagesFromMessagesDTO(messagesDTO);
+            return GetLatestMessages(parameters, out _);
         }
 
-        // Publish Message
-       public IMessage PublishMessage(string text, long recipientId)
+        public IEnumerable<IMessage> GetLatestMessages(IGetMessagesParameters parameters, out string cursor)
         {
-            return PublishMessage(text, new UserIdentifier(recipientId));
-        }
+            if (parameters == null)
+            {
+                throw new ArgumentNullException(nameof(parameters));
+            }
 
-        public IMessage PublishMessage(string text, string recipientUserName)
-        {
-            return PublishMessage(text, new UserIdentifier(recipientUserName));
+            // If we've been asked to fetch more than the maximum number of results that Twitter will return from
+            //  a single API request, we need to break this up into multiple requests
+            if (parameters.Count > TweetinviConsts.MESSAGE_GET_COUNT)
+            {
+                // Firstly, run a request for the maximum number of messages from the supplied cursor
+                //  Note: copy the request rather than just updating the count in case the caller intends to
+                //  reuse the parameters object for future requests
+                var thisReqParams = _getMessagesParametersFactory.Create();
+                thisReqParams.Count = TweetinviConsts.MESSAGE_GET_COUNT;
+                thisReqParams.Cursor = parameters.Cursor;
+
+                IEnumerable<IMessage> thisReqResults = GetLatestMessages(thisReqParams, out cursor);
+
+                // If there are more messages still available to be fetched from Twitter
```

Comment:
This is intentional.  

When you request say 10 messages from the Twitter API it will run your request for the most recent 10 internally and *then* filter out deleted ones. The order here is key as it means you may well get less than 10 results back to your request for the last 10 messages, and if there are any older messages, there will also be a cursor included.  
In that case, I thought it best to pass however many messages we have received from Twitter back to the user and not try and keep making requests until we get all 10 messages requested. Doing so could lead to an unexpectedly large number of API calls, burbning through all of the user's rate limits with them still thinking they;ve only made a single request.